tests/testthat/test-indentation_linter.R

```diff
@@ -0,0 +1,423 @@
+test_that("indentation linter flags unindented expressions", {
+  linter <- indentation_linter(indent = 2L)
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        i %% 2
+      })
+    "),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+       i %% 2  # indentation is only 1 character
+      })
+    "),
+    "Indentation",
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+          i %% 2
+      })
+    "),
+    NULL,
+    indentation_linter(indent = 4L)
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        i %% 2  # indentation is only 2 characters
+      })
+    "),
+    "Indentation",
+    indentation_linter(indent = 4L)
+  )
+
+  # ugly code, but still correctly indented
+  expect_lint(
+    trim_some("
+      list(
+           1,
+           2)
+    "),
+    NULL,
+    linter
+  )
+
+  # comments do not trigger hanging indent rule
```

Comment:
i don't see any cases when the comment is mis-indented, please add tests for this e.g.

```r
list(
 # comment at 1 space
  key = "correct spaced code"
)
```